File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        mapp1 = Counter(set(nums1))
        mapp2 = Counter(set(nums2))
        res = []
        for key in mapp1: 
            if key in mapp2: 
                res.extend([key] * mapp1[key])
        return res 

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        // vector<int> intersect; 
        // unordered_set<int> Final; 
         
        // for(int i = 0; i < nums1.size(); i ++){
        //     for(int j = 0; j < nums2.size(); j++){
        //         if (nums1[i] == nums2[j]){
        //             Final.insert(nums1[i]); 
        //         }
        //     }
        // }
        // vector<int> Final_vector(Final.begin(), Final.end());
        
        // return Final_vector; 

        unordered_set<int> set(nums1.begin(), nums1.end()); 
        unordered_set<int> store; 

        for(int num : nums2){
            if(set.count(num)){
                store.insert(num); 
            }
        }
        vector<int> Final(store.begin(), store.end()); 
        return Final; 
        
    }
};